src/components/molecules/Visualizer/Engine/ref.ts

```diff
@@ -49,6 +48,19 @@ export type MouseEventHandles = {
   onWheel: (fn: MouseEvents["wheel"]) => void;
 };
 
+type ClockEventHandler<R = void> = (cb: (clock: Clock) => void) => R;
+export type Clock = {
+  startTime: Date;
+  stopTime: Date;
+  currentTime: Date;
+  tick: () => Date;
+  shouldAnimate: boolean;
+  onTick: {
+    addEventListener: ClockEventHandler<() => void>;
+    removeEventListener: ClockEventHandler;
+  };
```

Comment:
To make the plugin API easier to understand, instead of exposing an event object, let's add a tick event to the plugin, similar to the cameramove event.

Add a new event "tick" to https://github.com/reearth/reearth-web/blob/main/src/components/molecules/Visualizer/Plugin/types.ts#L45 and implement rest

Reply:
This was fixed in [fix: use reearth event](https://github.com/reearth/reearth-web/pull/285/commits/8375e5eed05e6e89e92840f3cbd3096307e7fbb9)